## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#724fb7d0-ba49-461c-a407-d65213b3dd96) |
| mint-fs.sh | :x: [more...](#a2f21d9e-0beb-427c-81f9-a12d329f22eb) |
| mint-dist-erasure.sh | :x: [more...](#6251294d-62c7-44b1-a3c8-90895659e0cd) |
| mint-pools.sh | :x: [more...](#3c0f6400-1fec-452c-87da-48c24407d0f9) |
<a name="3c0f6400-1fec-452c-87da-48c24407d0f9"></a>
### 14289-959b13c/mint-pools.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s7:32251
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-m3wxpxrjbajxeo",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 272,
  "error": "RequestError: send request failed\ncaused by: Put \"http://minio-k8s7:32251/aws-sdk-go-test-m3wxpxrjbajxeo\": dial tcp 15.15.15.7:32251: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-java tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-m3wxpxrjbajxeo",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 272,
  "error": "RequestError: send request failed\ncaused by: Put \"http://minio-k8s7:32251/aws-sdk-go-test-m3wxpxrjbajxeo\": dial tcp 15.15.15.7:32251: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-php tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-m3wxpxrjbajxeo",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 272,
  "error": "RequestError: send request failed\ncaused by: Put \"http://minio-k8s7:32251/aws-sdk-go-test-m3wxpxrjbajxeo\": dial tcp 15.15.15.7:32251: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-ruby tests ... FAILED in 4 seconds
{
  "name": "aws-sdk-ruby",
  "function": "listBuckets",
  "args": {},
  "duration": 2112.29,
  "error": "Failed to open TCP connection to minio-k8s7:32251 (Connection refused - connect(2) for \"minio-k8s7\" port 32251)",
  "status": "FAIL"
}
(1/14) Running awscli tests ... FAILED in 7 seconds
{"name": "awscli", "duration": 5753, "function": "make_bucket\n", "status": "FAIL", "error": "Could not connect to the endpoint URL: "http://minio-k8s7:32251/awscli-mint-test-bucket-6778""}
(1/14) Running healthcheck tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": null,
  "duration": 1,
  "error": "Get \"http://minio-k8s7:32251/minio/health/live\": dial tcp 15.15.15.7:32251: connect: connection refused",
  "function": "testLivenessEndpoint",
  "message": "GET request failed",
  "name": "healthcheck",
  "status": "FAIL"
}
(1/14) Running mc tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": null,
  "duration": 1,
  "error": "Get \"http://minio-k8s7:32251/minio/health/live\": dial tcp 15.15.15.7:32251: connect: connection refused",
  "function": "testLivenessEndpoint",
  "message": "GET request failed",
  "name": "healthcheck",
  "status": "FAIL"
}
(1/14) Running minio-dotnet tests ... done in 2 seconds
(2/14) Running minio-go tests ... FAILED in 4 seconds
{
  "args": {
    "bucketName": "minio-go-test-56rsvf5jpak42lc9",
    "region": "eu-west-1"
  },
  "duration": 4337,
  "error": "Put \"http://minio-k8s7:32251/minio-go-test-56rsvf5jpak42lc9/\": dial tcp 15.15.15.7:32251: connect: connection refused",
  "function": "MakeBucket(bucketName, region)",
  "message": "MakeBucket failed",
  "name": "minio-go: testMakeBucketErrorV2",
  "status": "FAIL"
}
(2/14) Running minio-java tests ... FAILED in 1 seconds
{
  "name": "minio-java",
  "function": "makeBucket()",
  "args": "[basic check]",
  "duration": 52,
  "status": "FAIL",
  "error": "java.net.ConnectException: Failed to connect to minio-k8s7/15.15.15.7:32251 >>> [okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.kt:297), okhttp3.internal.connection.RealConnection.connect(RealConnection.kt:207), okhttp3.internal.connection.ExchangeFinder.findConnection(ExchangeFinder.kt:226), okhttp3.internal.connection.ExchangeFinder.findHealthyConnection(ExchangeFinder.kt:106), okhttp3.internal.connection.ExchangeFinder.find(ExchangeFinder.kt:74), okhttp3.internal.connection.RealCall.initExchange$okhttp(RealCall.kt:255), okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:32), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.kt:95), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:83), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:76), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:201), okhttp3.internal.connection.RealCall.execute(RealCall.kt:154), io.minio.S3Base.execute(S3Base.java:507), io.minio.MinioClient.makeBucket(MinioClient.java:1199), FunctionalTest.testBucketApi(FunctionalTest.java:448), FunctionalTest.testBucketApiCases(FunctionalTest.java:476), FunctionalTest.makeBucket(FunctionalTest.java:507), FunctionalTest.runBucketTests(FunctionalTest.java:3700), FunctionalTest.runTests(FunctionalTest.java:3776), FunctionalTest.runEndpointTests(FunctionalTest.java:3876), FunctionalTest.main(FunctionalTest.java:3996)]"
}
(2/14) Running minio-py tests ... FAILED in 7 seconds
{
  "name": "minio-py:test_make_bucket_default_region",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-65050c9c-b735-4225-a10c-f15e94968114",
    "location": "default value ('us-east-1')"
  },
  "message": "HTTPConnectionPool(host='minio-k8s7', port=32251): Max retries exceeded with url: /minio-py-test-65050c9c-b735-4225-a10c-f15e94968114 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fa474d7e520>: Failed to establish a new connection: [Errno 111] Connection refused'))",
  "error": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 174, in _new_conn\n    conn = connection.create_connection(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/connection.py\", line 95, in create_connection\n    raise err\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/connection.py\", line 85, in create_connection\n    sock.connect(sa)\nConnectionRefusedError: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 703, in urlopen\n    httplib_response = self._make_request(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 398, in _make_request\n    conn.request(method, url, **httplib_request_kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 239, in request\n    super(HTTPConnection, self).request(method, url, body=body, headers=headers)\n  File \"/usr/lib/python3.8/http/client.py\", line 1256, in request\n    self._send_request(method, url, body, headers, encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1302, in _send_request\n    self.endheaders(body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1251, in endheaders\n    self._send_output(message_body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1011, in _send_output\n    self.send(msg)\n  File \"/usr/lib/python3.8/http/client.py\", line 951, in send\n    self.connect()\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 205, in connect\n    conn = self._new_conn()\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 186, in _new_conn\n    raise NewConnectionError(\nurllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7fa474d7e520>: Failed to establish a new connection: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 168, in test_make_bucket_default_region\n    _call(log_entry, _CLIENT.make_bucket, bucket_name)\n  File \"/mint/run/core/minio-py/tests.py\", line 109, in _call\n    return func(*args, **kwargs)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 612, in make_bucket\n    self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 266, in _url_open\n    response = self._http.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/poolmanager.py\", line 375, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 813, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 813, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 813, in urlopen\n    return self.urlopen(\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 785, in urlopen\n    retries = retries.increment(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py\", line 592, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-k8s7', port=32251): Max retries exceeded with url: /minio-py-test-65050c9c-b735-4225-a10c-f15e94968114 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fa474d7e520>: Failed to establish a new connection: [Errno 111] Connection refused'))\n",
  "function": "make_bucket(bucket_name, location, object_lock)",
  "duration": 6025
}
(2/14) Running s3cmd tests ... FAILED in 3 seconds
{
  "name": "s3cmd",
  "duration": "225",
  "function": "test_make_bucket",
  "status": "FAIL",
  "error": "\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n    An unexpected error has occurred.\n  Please try reproducing the error using\n  the latest s3cmd code from the git master\n  branch found at:\n    https://github.com/s3tools/s3cmd\n  and have a look at the known issues list:\n    https://github.com/s3tools/s3cmd/wiki/Common-known-issues-and-their-solutions-(FAQ)\n  If the error persists, please report the\n  following lines (removing any private\n  info as necessary) to:\n   s3tools-bugs@lists.sourceforge.net\n\n\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n\nInvoked as: /usr/local/bin/s3cmd --config /tmp/.s3cmd-2088/s3cfg mb s3://s3cmd-test-bucket-7698\nProblem: <class 'ConnectionRefusedError: [Errno 111] Connection refused\nS3cmd:   2.2.0\npython:   3.8.10 (default, Nov 26 2021, 20:14:08) \n[GCC 9.3.0]\nenvironment LANG=C.UTF-8\n\nTraceback (most recent call last):\n  File \"/usr/local/bin/s3cmd\", line 3209, in <module>\n    rc = main()\n  File \"/usr/local/bin/s3cmd\", line 3106, in main\n    rc = cmd_func(args)\n  File \"/usr/local/bin/s3cmd\", line 260, in cmd_bucket_create\n    response = s3.bucket_create(uri.bucket(), cfg.bucket_location, cfg.extra_headers)\n  File \"/usr/local/lib/python3.8/dist-packages/S3/S3.py\", line 430, in bucket_create\n    response = self.send_request(request)\n  File \"/usr/local/lib/python3.8/dist-packages/S3/S3.py\", line 1480, in send_request\n    conn = ConnMan.get(self.get_hostname(resource['bucket']))\n  File \"/usr/local/lib/python3.8/dist-packages/S3/ConnMan.py\", line 284, in get\n    conn.c.connect()\n  File \"/usr/lib/python3.8/http/client.py\", line 922, in connect\n    self.sock = self._create_connection(\n  File \"/usr/lib/python3.8/socket.py\", line 808, in create_connection\n    raise err\n  File \"/usr/lib/python3.8/socket.py\", line 796, in create_connection\n    sock.connect(sa)\nConnectionRefusedError: [Errno 111] Connection refused\n\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n    An unexpected error has occurred.\n  Please try reproducing the error using\n  the latest s3cmd code from the git master\n  branch found at:\n    https://github.com/s3tools/s3cmd\n  and have a look at the known issues list:\n    https://github.com/s3tools/s3cmd/wiki/Common-known-issues-and-their-solutions-(FAQ)\n  If the error persists, please report the\n  above lines (removing any private\n  info as necessary) to:\n   s3tools-bugs@lists.sourceforge.net\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
}
(2/14) Running s3select tests ... FAILED in 6 seconds
{
  "name": "s3select:test_csv_input_quote_char",
  "function": "select_object_content(bucket_name, object_name, request)",
  "args": {
    "bucket_name": "s3select-test-c2504a16-ba98-4bc5-a31b-327023f4b936"
  },
  "duration": 6020,
  "message": "HTTPConnectionPool(host='minio-k8s7', port=32251): Max retries exceeded with url: /s3select-test-c2504a16-ba98-4bc5-a31b-327023f4b936 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f7434afa2e0>: Failed to establish a new connection: [Errno 111] Connection refused'))",
  "error": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 174, in _new_conn\n    conn = connection.create_connection(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/connection.py\", line 95, in create_connection\n    raise err\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/connection.py\", line 85, in create_connection\n    sock.connect(sa)\nConnectionRefusedError: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 703, in urlopen\n    httplib_response = self._make_request(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 398, in _make_request\n    conn.request(method, url, **httplib_request_kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 239, in request\n    super(HTTPConnection, self).request(method, url, body=body, headers=headers)\n  File \"/usr/lib/python3.8/http/client.py\", line 1256, in request\n    self._send_request(method, url, body, headers, encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1302, in _send_request\n    self.endheaders(body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1251, in endheaders\n    self._send_output(message_body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1011, in _send_output\n    self.send(msg)\n  File \"/usr/lib/python3.8/http/client.py\", line 951, in send\n    self.connect()\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 205, in connect\n    conn = self._new_conn()\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 186, in _new_conn\n    raise NewConnectionError(\nurllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7f7434afa2e0>: Failed to establish a new connection: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"./tests.py\", line 53, in main\n    test_csv_input_custom_quote_char(client, log_output)\n  File \"/mint/run/core/s3select/csv.py\", line 95, in test_csv_input_custom_quote_char\n    client.make_bucket(bucket_name)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 612, in make_bucket\n    self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 266, in _url_open\n    response = self._http.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/poolmanager.py\", line 375, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 813, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 813, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 813, in urlopen\n    return self.urlopen(\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 785, in urlopen\n    retries = retries.increment(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py\", line 592, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-k8s7', port=32251): Max retries exceeded with url: /s3select-test-c2504a16-ba98-4bc5-a31b-327023f4b936 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f7434afa2e0>: Failed to establish a new connection: [Errno 111] Connection refused'))\n",
  "status": "FAIL"
}
(2/14) Running versioning tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "versioning-test-pm9h0uev5ic1iy"
  },
  "duration": 278,
  "error": "RequestError: send request failed\ncaused by: Put \"http://minio-k8s7:32251/versioning-test-pm9h0uev5ic1iy\": dial tcp 15.15.15.7:32251: connect: connection refused",
  "function": "/mint/build/versioning/bucket.go:47: main.testMakeBucket",
  "message": "Versioning CreateBucket Failed",
  "name": "versioning",
  "status": "FAIL"
}

Executed 1 out of 14 tests successfully.
```
<a name="6251294d-62c7-44b1-a3c8-90895659e0cd"></a>
### 14289-959b13c/mint-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s8:30816
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 6 minutes and 6 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-knx9n4591xf01d",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 365209,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 16D2C68D86F8AD99, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-java tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-knx9n4591xf01d",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 365209,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 16D2C68D86F8AD99, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-php tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-knx9n4591xf01d",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 365209,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 16D2C68D86F8AD99, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-ruby tests ... FAILED in 4 seconds
{
  "name": "aws-sdk-ruby",
  "function": "listBuckets",
  "args": {},
  "duration": 2113.14,
  "error": "Server not initialized, please try again.",
  "status": "FAIL"
}
(1/14) Running awscli tests ... FAILED in 4 seconds
{
  "name": "awscli",
  "duration": 3205,
  "function": "make_bucket\n",
  "status": "FAIL",
  "error": "An error occurred (XMinioServerNotInitialized) when calling the CreateBucket operation (reached max retries: 4): Server not initialized, please try again."
}
(1/14) Running healthcheck tests ... done in 1 seconds
(2/14) Running mc tests ... FAILED in 0 seconds
(2/14) Running minio-dotnet tests ... done in 2 seconds
(3/14) Running minio-go tests ... FAILED in 5 seconds
{
  "args": {
    "bucketName": "minio-go-test-d1d1v99pg91ln3n5",
    "region": "eu-west-1"
  },
  "duration": 5435,
  "error": "Server not initialized, please try again.",
  "function": "MakeBucket(bucketName, region)",
  "message": "MakeBucket failed",
  "name": "minio-go: testMakeBucketErrorV2",
  "status": "FAIL"
}
(3/14) Running minio-java tests ... FAILED in 2 seconds
{
  "name": "minio-java",
  "function": "makeBucket()",
  "args": "[basic check]",
  "duration": 196,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = XMinioServerNotInitialized, message = Server not initialized, please try again., bucketName = minio-java-test-1bd7kut, objectName = null, resource = /minio-java-test-1bd7kut, requestId = 16D2C6921EB73235, hostId = null)\nrequest={method=PUT, url=http://minio-k8s8:30816/minio-java-test-1bd7kut, headers=Host: minio-k8s8:30816\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.3.5\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20220211T155904Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20220211/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=503, headers=Accept-Ranges: bytes\nContent-Length: 302\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nRetry-After: 120\nServer: MinIO\nStrict-Transport-Security: max-age=31536000; includeSubDomains\nVary: Origin\nVary: Accept-Encoding\nX-Amz-Request-Id: 16D2C6921EB73235\nX-Content-Type-Options: nosniff\nX-Xss-Protection: 1; mode=block\nDate: Fri, 11 Feb 2022 15:59:04 GMT\n}\n >>> [io.minio.S3Base.execute(S3Base.java:645), io.minio.MinioClient.makeBucket(MinioClient.java:1199), FunctionalTest.testBucketApi(FunctionalTest.java:448), FunctionalTest.testBucketApiCases(FunctionalTest.java:476), FunctionalTest.makeBucket(FunctionalTest.java:507), FunctionalTest.runBucketTests(FunctionalTest.java:3700), FunctionalTest.runTests(FunctionalTest.java:3776), FunctionalTest.runEndpointTests(FunctionalTest.java:3876), FunctionalTest.main(FunctionalTest.java:3996)]"
}
(3/14) Running minio-py tests ... FAILED in 10 minutes and 0 seconds
{
  "name": "minio-py:test_make_bucket_default_region",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-102bc5b0-0352-45fe-acda-37a44a842844",
    "location": "default value ('us-east-1')"
  },
  "message": "HTTPConnectionPool(host='minio-k8s8', port=30816): Max retries exceeded with url: /minio-py-test-102bc5b0-0352-45fe-acda-37a44a842844 (Caused by ResponseError('too many 503 error responses'))",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 168, in test_make_bucket_default_region\n    _call(log_entry, _CLIENT.make_bucket, bucket_name)\n  File \"/mint/run/core/minio-py/tests.py\", line 109, in _call\n    return func(*args, **kwargs)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 612, in make_bucket\n    self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 266, in _url_open\n    response = self._http.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/poolmanager.py\", line 375, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 876, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 876, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 876, in urlopen\n    return self.urlopen(\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 866, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py\", line 592, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-k8s8', port=30816): Max retries exceeded with url: /minio-py-test-102bc5b0-0352-45fe-acda-37a44a842844 (Caused by ResponseError('too many 503 error responses'))\n",
  "function": "make_bucket(bucket_name, location, object_lock)",
  "duration": 600404
}
(3/14) Running s3cmd tests ... FAILED in 46 seconds
{
  "name": "s3cmd",
  "duration": "45327",
  "function": "test_make_bucket",
  "status": "FAIL",
  "error": "WARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 3 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 6 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 9 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 12 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 15 sec...\nERROR: S3 error: 503 (XMinioServerNotInitialized): Server not initialized, please try again."
}
(3/14) Running s3select tests ... FAILED in 10 minutes and 1 seconds
{
  "name": "s3select:test_csv_input_quote_char",
  "function": "select_object_content(bucket_name, object_name, request)",
  "args": {
    "bucket_name": "s3select-test-d72d9167-a3b2-43ec-b482-1f1388189b6e"
  },
  "duration": 600525,
  "message": "HTTPConnectionPool(host='minio-k8s8', port=30816): Max retries exceeded with url: /s3select-test-d72d9167-a3b2-43ec-b482-1f1388189b6e (Caused by ResponseError('too many 503 error responses'))",
  "error": "Traceback (most recent call last):\n  File \"./tests.py\", line 53, in main\n    test_csv_input_custom_quote_char(client, log_output)\n  File \"/mint/run/core/s3select/csv.py\", line 95, in test_csv_input_custom_quote_char\n    client.make_bucket(bucket_name)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 612, in make_bucket\n    self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 266, in _url_open\n    response = self._http.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/poolmanager.py\", line 375, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 876, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 876, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 876, in urlopen\n    return self.urlopen(\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 866, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py\", line 592, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-k8s8', port=30816): Max retries exceeded with url: /s3select-test-d72d9167-a3b2-43ec-b482-1f1388189b6e (Caused by ResponseError('too many 503 error responses'))\n",
  "status": "FAIL"
}
(3/14) Running versioning tests ... FAILED in 6 minutes and 5 seconds
{
  "alert": "",
  "args": {
    "bucketName": "versioning-test-yxd4j4q0jwbbpv"
  },
  "duration": 365327,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 16D2C8099B80DCB5, host id: ",
  "function": "/mint/build/versioning/bucket.go:47: main.testMakeBucket",
  "message": "Versioning CreateBucket Failed",
  "name": "versioning",
  "status": "FAIL"
}

Executed 2 out of 14 tests successfully.
```
<a name="a2f21d9e-0beb-427c-81f9-a12d329f22eb"></a>
### 14289-959b13c/mint-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s2:30405
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-4lky1zv45adfe4",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 318,
  "error": "RequestError: send request failed\ncaused by: Put \"http://minio-k8s2:30405/aws-sdk-go-test-4lky1zv45adfe4\": dial tcp 15.15.15.2:30405: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-java tests ... FAILED in 3 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-4lky1zv45adfe4",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 318,
  "error": "RequestError: send request failed\ncaused by: Put \"http://minio-k8s2:30405/aws-sdk-go-test-4lky1zv45adfe4\": dial tcp 15.15.15.2:30405: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-php tests ... FAILED in 5 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-4lky1zv45adfe4",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 318,
  "error": "RequestError: send request failed\ncaused by: Put \"http://minio-k8s2:30405/aws-sdk-go-test-4lky1zv45adfe4\": dial tcp 15.15.15.2:30405: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-ruby tests ... FAILED in 5 seconds
{
  "name": "aws-sdk-ruby",
  "function": "listBuckets",
  "args": {},
  "duration": 2110.99,
  "error": "Failed to open TCP connection to minio-k8s2:30405 (Connection refused - connect(2) for \"minio-k8s2\" port 30405)",
  "status": "FAIL"
}
(1/14) Running awscli tests ... FAILED in 10 seconds
{"name": "awscli", "duration": 8897, "function": "make_bucket\n", "status": "FAIL", "error": "Could not connect to the endpoint URL: "http://minio-k8s2:30405/awscli-mint-test-bucket-30713""}
(1/14) Running healthcheck tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": null,
  "duration": 1,
  "error": "Get \"http://minio-k8s2:30405/minio/health/live\": dial tcp 15.15.15.2:30405: connect: connection refused",
  "function": "testLivenessEndpoint",
  "message": "GET request failed",
  "name": "healthcheck",
  "status": "FAIL"
}
(1/14) Running mc tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": null,
  "duration": 1,
  "error": "Get \"http://minio-k8s2:30405/minio/health/live\": dial tcp 15.15.15.2:30405: connect: connection refused",
  "function": "testLivenessEndpoint",
  "message": "GET request failed",
  "name": "healthcheck",
  "status": "FAIL"
}
(1/14) Running minio-dotnet tests ... done in 44 seconds
(2/14) Running minio-go tests ... FAILED in 10 seconds
{
  "args": {
    "bucketName": "minio-go-test-ijeozkvfe6vz913e",
    "region": "eu-west-1"
  },
  "duration": 10228,
  "error": "Put \"http://minio-k8s2:30405/minio-go-test-ijeozkvfe6vz913e/\": dial tcp 15.15.15.2:30405: connect: connection refused",
  "function": "MakeBucket(bucketName, region)",
  "message": "MakeBucket failed",
  "name": "minio-go: testMakeBucketErrorV2",
  "status": "FAIL"
}
(2/14) Running minio-java tests ... FAILED in 1 seconds
{
  "name": "minio-java",
  "function": "makeBucket()",
  "args": "[basic check]",
  "duration": 52,
  "status": "FAIL",
  "error": "java.net.ConnectException: Failed to connect to minio-k8s2/15.15.15.2:30405 >>> [okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.kt:297), okhttp3.internal.connection.RealConnection.connect(RealConnection.kt:207), okhttp3.internal.connection.ExchangeFinder.findConnection(ExchangeFinder.kt:226), okhttp3.internal.connection.ExchangeFinder.findHealthyConnection(ExchangeFinder.kt:106), okhttp3.internal.connection.ExchangeFinder.find(ExchangeFinder.kt:74), okhttp3.internal.connection.RealCall.initExchange$okhttp(RealCall.kt:255), okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:32), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.kt:95), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:83), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:76), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:201), okhttp3.internal.connection.RealCall.execute(RealCall.kt:154), io.minio.S3Base.execute(S3Base.java:507), io.minio.MinioClient.makeBucket(MinioClient.java:1199), FunctionalTest.testBucketApi(FunctionalTest.java:448), FunctionalTest.testBucketApiCases(FunctionalTest.java:476), FunctionalTest.makeBucket(FunctionalTest.java:507), FunctionalTest.runBucketTests(FunctionalTest.java:3700), FunctionalTest.runTests(FunctionalTest.java:3776), FunctionalTest.runEndpointTests(FunctionalTest.java:3876), FunctionalTest.main(FunctionalTest.java:3996)]"
}
(2/14) Running minio-py tests ... FAILED in 8 seconds
{
  "name": "minio-py:test_make_bucket_default_region",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-029a21c1-bbfc-47b8-b6d3-7d704d8d4b25",
    "location": "default value ('us-east-1')"
  },
  "message": "HTTPConnectionPool(host='minio-k8s2', port=30405): Max retries exceeded with url: /minio-py-test-029a21c1-bbfc-47b8-b6d3-7d704d8d4b25 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f2240a0d550>: Failed to establish a new connection: [Errno 111] Connection refused'))",
  "error": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 174, in _new_conn\n    conn = connection.create_connection(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/connection.py\", line 95, in create_connection\n    raise err\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/connection.py\", line 85, in create_connection\n    sock.connect(sa)\nConnectionRefusedError: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 703, in urlopen\n    httplib_response = self._make_request(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 398, in _make_request\n    conn.request(method, url, **httplib_request_kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 239, in request\n    super(HTTPConnection, self).request(method, url, body=body, headers=headers)\n  File \"/usr/lib/python3.8/http/client.py\", line 1256, in request\n    self._send_request(method, url, body, headers, encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1302, in _send_request\n    self.endheaders(body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1251, in endheaders\n    self._send_output(message_body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1011, in _send_output\n    self.send(msg)\n  File \"/usr/lib/python3.8/http/client.py\", line 951, in send\n    self.connect()\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 205, in connect\n    conn = self._new_conn()\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 186, in _new_conn\n    raise NewConnectionError(\nurllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7f2240a0d550>: Failed to establish a new connection: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 168, in test_make_bucket_default_region\n    _call(log_entry, _CLIENT.make_bucket, bucket_name)\n  File \"/mint/run/core/minio-py/tests.py\", line 109, in _call\n    return func(*args, **kwargs)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 612, in make_bucket\n    self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 266, in _url_open\n    response = self._http.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/poolmanager.py\", line 375, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 813, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 813, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 813, in urlopen\n    return self.urlopen(\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 785, in urlopen\n    retries = retries.increment(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py\", line 592, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-k8s2', port=30405): Max retries exceeded with url: /minio-py-test-029a21c1-bbfc-47b8-b6d3-7d704d8d4b25 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f2240a0d550>: Failed to establish a new connection: [Errno 111] Connection refused'))\n",
  "function": "make_bucket(bucket_name, location, object_lock)",
  "duration": 8056
}
(2/14) Running s3cmd tests ... FAILED in 0 seconds
{
  "name": "s3cmd",
  "duration": "182",
  "function": "test_make_bucket",
  "status": "FAIL",
  "error": "\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n    An unexpected error has occurred.\n  Please try reproducing the error using\n  the latest s3cmd code from the git master\n  branch found at:\n    https://github.com/s3tools/s3cmd\n  and have a look at the known issues list:\n    https://github.com/s3tools/s3cmd/wiki/Common-known-issues-and-their-solutions-(FAQ)\n  If the error persists, please report the\n  following lines (removing any private\n  info as necessary) to:\n   s3tools-bugs@lists.sourceforge.net\n\n\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n\nInvoked as: /usr/local/bin/s3cmd --config /tmp/.s3cmd-28299/s3cfg mb s3://s3cmd-test-bucket-31711\nProblem: <class 'ConnectionRefusedError: [Errno 111] Connection refused\nS3cmd:   2.2.0\npython:   3.8.10 (default, Nov 26 2021, 20:14:08) \n[GCC 9.3.0]\nenvironment LANG=C.UTF-8\n\nTraceback (most recent call last):\n  File \"/usr/local/bin/s3cmd\", line 3209, in <module>\n    rc = main()\n  File \"/usr/local/bin/s3cmd\", line 3106, in main\n    rc = cmd_func(args)\n  File \"/usr/local/bin/s3cmd\", line 260, in cmd_bucket_create\n    response = s3.bucket_create(uri.bucket(), cfg.bucket_location, cfg.extra_headers)\n  File \"/usr/local/lib/python3.8/dist-packages/S3/S3.py\", line 430, in bucket_create\n    response = self.send_request(request)\n  File \"/usr/local/lib/python3.8/dist-packages/S3/S3.py\", line 1480, in send_request\n    conn = ConnMan.get(self.get_hostname(resource['bucket']))\n  File \"/usr/local/lib/python3.8/dist-packages/S3/ConnMan.py\", line 284, in get\n    conn.c.connect()\n  File \"/usr/lib/python3.8/http/client.py\", line 922, in connect\n    self.sock = self._create_connection(\n  File \"/usr/lib/python3.8/socket.py\", line 808, in create_connection\n    raise err\n  File \"/usr/lib/python3.8/socket.py\", line 796, in create_connection\n    sock.connect(sa)\nConnectionRefusedError: [Errno 111] Connection refused\n\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n    An unexpected error has occurred.\n  Please try reproducing the error using\n  the latest s3cmd code from the git master\n  branch found at:\n    https://github.com/s3tools/s3cmd\n  and have a look at the known issues list:\n    https://github.com/s3tools/s3cmd/wiki/Common-known-issues-and-their-solutions-(FAQ)\n  If the error persists, please report the\n  above lines (removing any private\n  info as necessary) to:\n   s3tools-bugs@lists.sourceforge.net\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
}
(2/14) Running s3select tests ... FAILED in 7 seconds
{
  "name": "s3select:test_csv_input_quote_char",
  "function": "select_object_content(bucket_name, object_name, request)",
  "args": {
    "bucket_name": "s3select-test-6db54078-750a-456c-8e38-239dc55dbe99"
  },
  "duration": 7030,
  "message": "HTTPConnectionPool(host='minio-k8s2', port=30405): Max retries exceeded with url: /s3select-test-6db54078-750a-456c-8e38-239dc55dbe99 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f7320dda2b0>: Failed to establish a new connection: [Errno 111] Connection refused'))",
  "error": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 174, in _new_conn\n    conn = connection.create_connection(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/connection.py\", line 95, in create_connection\n    raise err\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/connection.py\", line 85, in create_connection\n    sock.connect(sa)\nConnectionRefusedError: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 703, in urlopen\n    httplib_response = self._make_request(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 398, in _make_request\n    conn.request(method, url, **httplib_request_kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 239, in request\n    super(HTTPConnection, self).request(method, url, body=body, headers=headers)\n  File \"/usr/lib/python3.8/http/client.py\", line 1256, in request\n    self._send_request(method, url, body, headers, encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1302, in _send_request\n    self.endheaders(body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1251, in endheaders\n    self._send_output(message_body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1011, in _send_output\n    self.send(msg)\n  File \"/usr/lib/python3.8/http/client.py\", line 951, in send\n    self.connect()\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 205, in connect\n    conn = self._new_conn()\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 186, in _new_conn\n    raise NewConnectionError(\nurllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7f7320dda2b0>: Failed to establish a new connection: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"./tests.py\", line 53, in main\n    test_csv_input_custom_quote_char(client, log_output)\n  File \"/mint/run/core/s3select/csv.py\", line 95, in test_csv_input_custom_quote_char\n    client.make_bucket(bucket_name)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 612, in make_bucket\n    self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 266, in _url_open\n    response = self._http.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/poolmanager.py\", line 375, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 813, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 813, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 813, in urlopen\n    return self.urlopen(\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 785, in urlopen\n    retries = retries.increment(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py\", line 592, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-k8s2', port=30405): Max retries exceeded with url: /s3select-test-6db54078-750a-456c-8e38-239dc55dbe99 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f7320dda2b0>: Failed to establish a new connection: [Errno 111] Connection refused'))\n",
  "status": "FAIL"
}
(2/14) Running versioning tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "versioning-test-wof9zhmqszgszq"
  },
  "duration": 345,
  "error": "RequestError: send request failed\ncaused by: Put \"http://minio-k8s2:30405/versioning-test-wof9zhmqszgszq\": dial tcp 15.15.15.2:30405: connect: connection refused",
  "function": "/mint/build/versioning/bucket.go:47: main.testMakeBucket",
  "message": "Versioning CreateBucket Failed",
  "name": "versioning",
  "status": "FAIL"
}

Executed 1 out of 14 tests successfully.
```
<a name="724fb7d0-ba49-461c-a407-d65213b3dd96"></a>
### 14289-959b13c/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s6:32206
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 6 minutes and 6 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-u0wi2yebzy32ra",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 366232,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 16D2C68DD09E73C5, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-java tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-u0wi2yebzy32ra",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 366232,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 16D2C68DD09E73C5, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-php tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-u0wi2yebzy32ra",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 366232,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 16D2C68DD09E73C5, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-ruby tests ... FAILED in 4 seconds
{
  "name": "aws-sdk-ruby",
  "function": "listBuckets",
  "args": {},
  "duration": 2116.09,
  "error": "Server not initialized, please try again.",
  "status": "FAIL"
}
(1/14) Running awscli tests ... FAILED in 13 seconds
{
  "name": "awscli",
  "duration": 11956,
  "function": "make_bucket\n",
  "status": "FAIL",
  "error": "An error occurred (XMinioServerNotInitialized) when calling the CreateBucket operation (reached max retries: 4): Server not initialized, please try again."
}
(1/14) Running healthcheck tests ... done in 1 seconds
(2/14) Running mc tests ... FAILED in 0 seconds
(2/14) Running minio-dotnet tests ... done in 2 seconds
(3/14) Running minio-go tests ... FAILED in 5 seconds
{
  "args": {
    "bucketName": "minio-go-test-ofeo3b1ly2gxsq2l",
    "region": "eu-west-1"
  },
  "duration": 4638,
  "error": "Server not initialized, please try again.",
  "function": "MakeBucket(bucketName, region)",
  "message": "MakeBucket failed",
  "name": "minio-go: testMakeBucketErrorV2",
  "status": "FAIL"
}
(3/14) Running minio-java tests ... FAILED in 1 seconds
{
  "name": "minio-java",
  "function": "makeBucket()",
  "args": "[basic check]",
  "duration": 195,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = XMinioServerNotInitialized, message = Server not initialized, please try again., bucketName = minio-java-test-278lrv, objectName = null, resource = /minio-java-test-278lrv, requestId = 16D2C69446D29B16, hostId = null)\nrequest={method=PUT, url=http://minio-k8s6:32206/minio-java-test-278lrv, headers=Host: minio-k8s6:32206\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.3.5\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20220211T155914Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20220211/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=503, headers=Accept-Ranges: bytes\nContent-Length: 300\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nRetry-After: 120\nServer: MinIO\nStrict-Transport-Security: max-age=31536000; includeSubDomains\nVary: Origin\nVary: Accept-Encoding\nX-Amz-Request-Id: 16D2C69446D29B16\nX-Content-Type-Options: nosniff\nX-Xss-Protection: 1; mode=block\nDate: Fri, 11 Feb 2022 15:59:14 GMT\n}\n >>> [io.minio.S3Base.execute(S3Base.java:645), io.minio.MinioClient.makeBucket(MinioClient.java:1199), FunctionalTest.testBucketApi(FunctionalTest.java:448), FunctionalTest.testBucketApiCases(FunctionalTest.java:476), FunctionalTest.makeBucket(FunctionalTest.java:507), FunctionalTest.runBucketTests(FunctionalTest.java:3700), FunctionalTest.runTests(FunctionalTest.java:3776), FunctionalTest.runEndpointTests(FunctionalTest.java:3876), FunctionalTest.main(FunctionalTest.java:3996)]"
}
(3/14) Running minio-py tests ... FAILED in 10 minutes and 0 seconds
{
  "name": "minio-py:test_make_bucket_default_region",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-95008c5e-b9d8-46e8-87a2-5e6b74eafb72",
    "location": "default value ('us-east-1')"
  },
  "message": "HTTPConnectionPool(host='minio-k8s6', port=32206): Max retries exceeded with url: /minio-py-test-95008c5e-b9d8-46e8-87a2-5e6b74eafb72 (Caused by ResponseError('too many 503 error responses'))",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 168, in test_make_bucket_default_region\n    _call(log_entry, _CLIENT.make_bucket, bucket_name)\n  File \"/mint/run/core/minio-py/tests.py\", line 109, in _call\n    return func(*args, **kwargs)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 612, in make_bucket\n    self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 266, in _url_open\n    response = self._http.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/poolmanager.py\", line 375, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 876, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 876, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 876, in urlopen\n    return self.urlopen(\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 866, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py\", line 592, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-k8s6', port=32206): Max retries exceeded with url: /minio-py-test-95008c5e-b9d8-46e8-87a2-5e6b74eafb72 (Caused by ResponseError('too many 503 error responses'))\n",
  "function": "make_bucket(bucket_name, location, object_lock)",
  "duration": 600365
}
(3/14) Running s3cmd tests ... FAILED in 45 seconds
{
  "name": "s3cmd",
  "duration": "45307",
  "function": "test_make_bucket",
  "status": "FAIL",
  "error": "WARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 3 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 6 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 9 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 12 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 15 sec...\nERROR: S3 error: 503 (XMinioServerNotInitialized): Server not initialized, please try again."
}
(3/14) Running s3select tests ... FAILED in 10 minutes and 1 seconds
{
  "name": "s3select:test_csv_input_quote_char",
  "function": "select_object_content(bucket_name, object_name, request)",
  "args": {
    "bucket_name": "s3select-test-d2d6931d-c62d-4dcd-979f-ae77231935e5"
  },
  "duration": 600354,
  "message": "HTTPConnectionPool(host='minio-k8s6', port=32206): Max retries exceeded with url: /s3select-test-d2d6931d-c62d-4dcd-979f-ae77231935e5 (Caused by ResponseError('too many 503 error responses'))",
  "error": "Traceback (most recent call last):\n  File \"./tests.py\", line 53, in main\n    test_csv_input_custom_quote_char(client, log_output)\n  File \"/mint/run/core/s3select/csv.py\", line 95, in test_csv_input_custom_quote_char\n    client.make_bucket(bucket_name)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 612, in make_bucket\n    self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 266, in _url_open\n    response = self._http.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/poolmanager.py\", line 375, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 876, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 876, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 876, in urlopen\n    return self.urlopen(\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 866, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py\", line 592, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-k8s6', port=32206): Max retries exceeded with url: /s3select-test-d2d6931d-c62d-4dcd-979f-ae77231935e5 (Caused by ResponseError('too many 503 error responses'))\n",
  "status": "FAIL"
}
(3/14) Running versioning tests ... FAILED in 6 minutes and 5 seconds
{
  "alert": "",
  "args": {
    "bucketName": "versioning-test-gj2nnp3qpgmqvm"
  },
  "duration": 365366,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 16D2C80BB7AA366B, host id: ",
  "function": "/mint/build/versioning/bucket.go:47: main.testMakeBucket",
  "message": "Versioning CreateBucket Failed",
  "name": "versioning",
  "status": "FAIL"
}

Executed 2 out of 14 tests successfully.
```
Deleting image on docker hub
Deleting image locally

--- a/cmd/config-current.go
+++ b/cmd/config-current.go
@@ -657,11 +657,12 @@ func applyDynamicConfig(ctx context.Context, objAPI ObjectLayer, s config.Config
 		logger.LogIf(ctx, fmt.Errorf("Unable to load logger webhook config: %w", err))
 	}
 	userAgent := getUserAgent(getMinioMode())
-	for _, l := range loggerCfg.HTTP {
+	for n, l := range loggerCfg.HTTP {
 		if l.Enabled {
 			l.LogOnce = logger.LogOnceIf
 			l.UserAgent = userAgent
 			l.Transport = NewGatewayHTTPTransportWithClientCerts(l.ClientCert, l.ClientKey)
+			loggerCfg.HTTP[n] = l
 		}
 	}
 	err = logger.UpdateTargets(loggerCfg)

--- a/cmd/consolelogger.go
+++ b/cmd/consolelogger.go
@@ -150,6 +150,10 @@ func (sys *HTTPConsoleLoggerSys) Content() (logs []log.Entry) {
 	return
 }
 
+// Cancel - cancels the target
+func (sys *HTTPConsoleLoggerSys) Cancel() {
+}
+
 // Send log message 'e' to console and publish to console
 // log pubsub system
 func (sys *HTTPConsoleLoggerSys) Send(e interface{}, logKind string) error {

--- a/cmd/data-update-tracker_test.go
+++ b/cmd/data-update-tracker_test.go
@@ -55,6 +55,9 @@ func (t *testingLogger) Init() error {
 	return nil
 }
 
+func (t *testingLogger) Cancel() {
+}
+
 func (t *testingLogger) Send(entry interface{}, errKind string) error {
 	t.mu.Lock()
 	defer t.mu.Unlock()

--- a/internal/config/config.go
+++ b/internal/config/config.go
@@ -167,7 +167,6 @@ var SubSystemsDynamic = set.CreateStringSet(
 	HealSubSys,
 	SubnetSubSys,
 	LoggerWebhookSubSys,
-	AuditWebhookSubSys,
 )
 
 // SubSystemsSingleTargets - subsystems which only support single target.

--- a/internal/logger/target/http/http.go
+++ b/internal/logger/target/http/http.go
@@ -58,6 +58,9 @@ type Target struct {
 	// Channel of log entries
 	logCh chan interface{}
 
+	// Cancellation function for the target
+	CancelFn context.CancelFunc
+
 	config Config
 }
 
@@ -121,54 +124,62 @@ func acceptedResponseStatusCode(code int) bool {
 }
 
 func (h *Target) startHTTPLogger() {
+	ctx, cancelFn := context.WithCancel(context.Background())
 	// Create a routine which sends json logs received
 	// from an internal channel.
 	go func() {
-		for entry := range h.logCh {
-			logJSON, err := json.Marshal(&entry)
-			if err != nil {
-				continue
-			}
+		for {
+			select {
+			case <-ctx.Done():
+				close(h.logCh)
+				return
+			case entry := <-h.logCh:
+				logJSON, err := json.Marshal(&entry)
+				if err != nil {
+					continue
+				}
 
-			ctx, cancel := context.WithTimeout(context.Background(), webhookCallTimeout)
-			req, err := http.NewRequestWithContext(ctx, http.MethodPost,
-				h.config.Endpoint, bytes.NewReader(logJSON))
-			if err != nil {
-				h.config.LogOnce(ctx, fmt.Errorf("%s returned '%w', please check your endpoint configuration", h.config.Endpoint, err), h.config.Endpoint)
+				ctx, cancel := context.WithTimeout(context.Background(), webhookCallTimeout)
+				req, err := http.NewRequestWithContext(ctx, http.MethodPost,
+					h.config.Endpoint, bytes.NewReader(logJSON))
+				if err != nil {
+					h.config.LogOnce(ctx, fmt.Errorf("%s returned '%w', please check your endpoint configuration", h.config.Endpoint, err), h.config.Endpoint)
+					cancel()
+					continue
+				}
+				req.Header.Set(xhttp.ContentType, "application/json")
+
+				// Set user-agent to indicate MinIO release
+				// version to the configured log endpoint
+				req.Header.Set("User-Agent", h.config.UserAgent)
+
+				if h.config.AuthToken != "" {
+					req.Header.Set("Authorization", h.config.AuthToken)
+				}
+
+				client := http.Client{Transport: h.config.Transport}
+				resp, err := client.Do(req)
 				cancel()
-				continue
-			}
-			req.Header.Set(xhttp.ContentType, "application/json")
+				if err != nil {
+					h.config.LogOnce(ctx, fmt.Errorf("%s returned '%w', please check your endpoint configuration", h.config.Endpoint, err), h.config.Endpoint)
+					continue
+				}
 
-			// Set user-agent to indicate MinIO release
-			// version to the configured log endpoint
-			req.Header.Set("User-Agent", h.config.UserAgent)
+				// Drain any response.
+				xhttp.DrainBody(resp.Body)
 
-			if h.config.AuthToken != "" {
-				req.Header.Set("Authorization", h.config.AuthToken)
-			}
-
-			client := http.Client{Transport: h.config.Transport}
-			resp, err := client.Do(req)
-			cancel()
-			if err != nil {
-				h.config.LogOnce(ctx, fmt.Errorf("%s returned '%w', please check your endpoint configuration", h.config.Endpoint, err), h.config.Endpoint)
-				continue
-			}
-
-			// Drain any response.
-			xhttp.DrainBody(resp.Body)
-
-			if !acceptedResponseStatusCode(resp.StatusCode) {
-				switch resp.StatusCode {
-				case http.StatusForbidden:
-					h.config.LogOnce(ctx, fmt.Errorf("%s returned '%s', please check if your auth token is correctly set", h.config.Endpoint, resp.Status), h.config.Endpoint)
-				default:
-					h.config.LogOnce(ctx, fmt.Errorf("%s returned '%s', please check your endpoint configuration", h.config.Endpoint, resp.Status), h.config.Endpoint)
+				if !acceptedResponseStatusCode(resp.StatusCode) {
+					switch resp.StatusCode {
+					case http.StatusForbidden:
+						h.config.LogOnce(ctx, fmt.Errorf("%s returned '%s', please check if your auth token is correctly set", h.config.Endpoint, resp.Status), h.config.Endpoint)
+					default:
+						h.config.LogOnce(ctx, fmt.Errorf("%s returned '%s', please check your endpoint configuration", h.config.Endpoint, resp.Status), h.config.Endpoint)
+					}
 				}
 			}
 		}
 	}()
+	h.CancelFn = cancelFn
 }
 
 // New initializes a new logger target which
@@ -193,4 +204,9 @@ func (h *Target) Send(entry interface{}, errKind string) error {
 	}
 
 	return nil
+}
+
+// Cancel - cancels the target
+func (h *Target) Cancel() {
+	h.CancelFn()
 }

--- a/internal/logger/target/kafka/kafka.go
+++ b/internal/logger/target/kafka/kafka.go
@@ -197,6 +197,10 @@ func (h *Target) Init() error {
 	return nil
 }
 
+// Cancel - cancels the target
+func (h *Target) Cancel() {
+}
+
 // New initializes a new logger target which
 // sends log over http to the specified endpoint
 func New(config Config) *Target {

--- a/internal/logger/targets.go
+++ b/internal/logger/targets.go
@@ -31,6 +31,7 @@ type Target interface {
 	String() string
 	Endpoint() string
 	Init() error
+	Cancel()
 	Send(entry interface{}, errKind string) error
 }
 
@@ -111,18 +112,31 @@ func AddTarget(t Target) error {
 	return nil
 }
 
-// UpdateTargets swaps targets with newly loaded ones from the cfg
-func UpdateTargets(cfg Config) error {
-	updated := []Target{}
+func cancelAllTargets() {
+	for _, tgt := range Targets() {
+		tgt.Cancel()
+	}
+}
 
+func initTargets(cfg Config) (tgts []Target, err error) {
 	for _, l := range cfg.HTTP {
 		if l.Enabled {
 			t := http.New(l)
-			if err := t.Init(); err != nil {
-				return err
+			if err = t.Init(); err != nil {
+				return tgts, err
 			}
-			updated = append(updated, t)
+			tgts = append(tgts, t)
 		}
+	}
+	return tgts, err
+}
+
+// UpdateTargets swaps targets with newly loaded ones from the cfg
+func UpdateTargets(cfg Config) error {
+	cancelAllTargets() // cancel running targets
+	updated, err := initTargets(cfg)
+	if err != nil {
+		return err
 	}
 
 	swapMu.Lock()

